src/background/providers/hcaptcha.ts

```diff
@@ -24,19 +81,179 @@ export class HcaptchaProvider implements Provider {
         this.callbacks.updateIcon(this.getBadgeText());
     }
 
+    private handleUrl(url: URL) {
+        const reqUrl = url.origin + url.pathname;
+        const isIssuerUrl = !!reqUrl.match(ISSUER_REGEX);
+
+        let isSpendUrl;
+        if (HCAPTCHA_URLS.some((url: string) => reqUrl === url)) {
+            isSpendUrl = false;
+        } else {
+            isSpendUrl = !!reqUrl.match(SPEND_REGEX);
+        }
+
+        return {
+            reqUrl,
+            isIssuerUrl,
+            isSpendUrl,
+        };
+    }
+
+    private async getCommitment(version: string): Promise<{ G: string; H: string }> {
+        const keyPrefix = 'commitment-';
+        const cached = this.storage.getItem(`${keyPrefix}${version}`);
+        if (cached !== null) {
+            return JSON.parse(cached);
+        }
+
+        interface Response {
+            HC: { [version: string]: { G: string; H: string; expiry: string; sig: string } };
+        }
+
+        // Download the commitment
+        const { data } = await axios.get<Response>(COMMITMENT_URL);
+        const commitment = data.HC[version as string];
+        if (commitment === undefined) {
+            throw new Error(`No commitment for the version ${version} is found`);
+        }
+
+        // Cache.
+        const item = {
+            G: commitment.G || voprf.sec1EncodeToBase64(voprf.getActiveECSettings().curve.G, false),
+            H: commitment.H,
+        };
+        this.storage.setItem(`${keyPrefix}${version}`, JSON.stringify(item));
+        return item;
+    }
+
+    private issue(url: string) {
+        (async () => {
+            const newTokens = Array.from(Array(NUM_OF_REQ_TOKENS).keys()).map(() => new Token());
```

Comment:
improves readability
```suggestion
            const newTokens = Array(NUM_OF_REQ_TOKENS).fill(new Token());
```